static/app/views/insights/common/components/spanDescription.tsx

```diff
@@ -47,26 +52,66 @@ export function DatabaseSpanDescription({
 }: Omit<Props, 'op'>) {
   const navigate = useNavigate();
   const location = useLocation();
+  const useEap = useInsightsEap();
+  const {projects} = useProjects();
+  const organization = useOrganization();
 
   const {data: indexedSpans, isFetching: areIndexedSpansLoading} = useSpansIndexed(
     {
       search: MutableSearch.fromQueryObject({'span.group': groupId}),
       limit: 1,
       fields: [
         SpanIndexedField.PROJECT_ID,
-        SpanIndexedField.TRANSACTION_ID,
+        SpanIndexedField.TRANSACTION_ID, // TODO: remove this with `useInsightsEap`, it's only needed to get the full event when eap is off
         SpanIndexedField.SPAN_DESCRIPTION,
+        SpanIndexedField.DB_SYSTEM,
+        SpanIndexedField.CODE_FILEPATH,
+        SpanIndexedField.CODE_LINENO,
+        SpanIndexedField.CODE_FUNCTION,
+        SpanIndexedField.SDK_NAME,
+        SpanIndexedField.SDK_VERSION,
+        SpanIndexedField.RELEASE,
       ],
+      sorts: [{field: SpanIndexedField.CODE_FILEPATH, kind: 'desc'}],
     },
     'api.starfish.span-description'
   );
   const indexedSpan = indexedSpans?.[0];
 
+  const project = projects.find(p => p.id === indexedSpan?.project_id?.toString());
+
+  const {data: eventDetailsData} = useEventDetails({
+    eventId: indexedSpan?.['transaction.id'],
+    projectSlug: project?.slug,
+  });
+
+  const {data: release} = useRelease({
+    orgSlug: organization.slug,
+    projectSlug: project?.slug ?? '',
+    releaseVersion: indexedSpan?.release ?? '',
+    enabled: useEap,
+  });
+
+  const sdk =
+    indexedSpan?.['sdk.name'] && indexedSpan?.['sdk.version']
+      ? {
+          name: indexedSpan?.['sdk.name'],
+          version: indexedSpan?.['sdk.version'],
+        }
+      : undefined;
+
+  const event = useEap
+    ? {
+        release,
+        sdk,
+      }
+    : eventDetailsData;
```

Comment:
It looks like `buildStacktraceLinkQuery` needs more parameters than this. It wants a platform, the event ID, etc. Are those things not needed to construct a working source code URL?